File: app/src/main/java/com/hinterlong/kevin/swishticker/ui/modules/TeamDetailActivity.java
package com.hinterlong.kevin.swishticker.ui.modules;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;

import com.hinterlong.kevin.swishticker.QueryEngine;
import com.hinterlong.kevin.swishticker.R;
import com.hinterlong.kevin.swishticker.data.Team;

import androidx.appcompat.app.AppCompatActivity;
import timber.log.Timber;

public class TeamDetailActivity extends AppCompatActivity {
    public static final String TEAM_ID = "TEAM_ID";
    private Team team;

    public static void withTeam(Context context, int teamId) {
        Intent intent = new Intent(context, TeamDetailActivity.class);
        intent.putExtra(TEAM_ID, teamId);
        context.startActivity(intent);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_team_detail_view);

        Intent intent = getIntent();
        int teamId = intent.getIntExtra(TEAM_ID, -1);
        if (teamId == -1) {
            Timber.e("No team id was provided");
        } else {
            team = QueryEngine.getInstance().getTeam(teamId);
        }
        setTitle(team.getName());
    }
}

File: app/src/main/java/com/hinterlong/kevin/swishticker/ui/modules/TeamDetailActivity.kt
package com.hinterlong.kevin.swishticker.ui.modules

import android.content.Context
import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.hinterlong.kevin.swishticker.R
import com.hinterlong.kevin.swishticker.service.AppDatabase
import com.hinterlong.kevin.swishticker.service.data.Team
import kotlinx.android.synthetic.main.toolbar.*

class TeamDetailActivity : AppCompatActivity() {
    private lateinit var team: Team

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_team_detail_view)

        val intent = intent
        val teamId = intent.getLongExtra(TEAM_ID, 0)
        team = AppDatabase.getInstance(this).teamDao().getTeam(teamId)

        val teamFragment = MyTeamFragment()
        teamFragment.arguments = Bundle().apply { putLong(MyTeamFragment.OVERRIDE_DEFAULT_TEAM, teamId) }
        supportFragmentManager.beginTransaction().add(R.id.contentFrame, teamFragment).commit()

        setSupportActionBar(toolbar)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        title = team.name
    }

    override fun onSupportNavigateUp(): Boolean {
        onBackPressed()
        return true
    }

    companion object {
        const val TEAM_ID = "TEAM_ID"

        fun withTeam(context: Context, teamId: Long) {
            val intent = Intent(context, TeamDetailActivity::class.java)
            intent.putExtra(TEAM_ID, teamId)
            context.startActivity(intent)
        }
    }
}
